spec/app/domain/services/update_course_excluded_exercises/service_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+RSpec.describe Services::UpdateCourseExcludedExercises::Service do
+  let(:service) { Services::UpdateCourseExcludedExercises::Service.new }
+
+  let(:action) { service.process(
+    course_uuid:      given_course_uuid,
+    sequence_number:  given_sequence_number,
+    exclusions:       given_exclusions
+  ) }
+
+  let(:given_exclusions) {
+    (given_specific_version_exclusions + given_any_version_exclusions).shuffle
+  }
+
+  let(:given_specific_version_exclusions) {
+    (number_of_exclusions - given_any_version_exclusions.length).times.map{
+      exercise_uuid = SecureRandom.uuid.to_s
+
+      given_exclusion_uuids.push(exercise_uuid)
+      { 'exercise_uuid' => exercise_uuid }
+    }
+  }
+
+  let(:given_any_version_exclusions) {
+    (Kernel::rand(number_of_exclusions)).times.map{
+      exercise_group_uuid = SecureRandom.uuid.to_s
+
+      given_exclusion_uuids.push(exercise_group_uuid)
+      { 'exercise_group_uuid' => exercise_group_uuid }
+    }
+  }
+
+  let(:given_course_uuid)     { SecureRandom.uuid.to_s }
+  let(:given_sequence_number) { Kernel::rand(10) }
+  let(:number_of_exclusions)  { 10 }
+  let(:given_exclusion_uuids) { [] }
+
+  context "when a previously non-existing Course uuid is given" do
+    it "raises error" do
+      expect{action}.to raise_error(Errors::AppUnprocessableError)
+    end
+  end
+
+  context "when previously-existing Course uuid is given" do
+    before(:each) do
+      create(:course, uuid: given_course_uuid)
+    end
+
+    context "and when excluded exercises are given" do
```

Comment:
There is apparently a missing case to cover when excluded exercises are not given.